File: Examples/ClassDiagram/Dart/song.dart
import 'media_item.dart';
import 'playable.dart';

/// A single track by an artist.
class Song extends MediaItem {
  Song(String title, double duration, Genre genre, this.artist, [this.album])
      : super(title, duration, genre);

  final String artist;
  final String? album;

  @override
  void play() {
    print('♪ $artist — $title');
  }
}

File: Examples/ClassDiagram/TypeScript/Song.ts
import { MediaItem } from "./MediaItem";
import { Genre } from "./Playable";

/** A single track by an artist. */
export class Song extends MediaItem {
    constructor(
        title: string,
        duration: number,
        genre: Genre,
        readonly artist: string,
        readonly album?: string,
    ) {
        super(title, duration, genre);
    }

    override play(): void {
        console.log(`♪ ${this.artist} — ${this.title}`);
    }
}
